gef-ui/server/services/__tests__/api.test.js

```diff
@@ -240,22 +242,108 @@ describe('deleteFacility()', () => {
 
-  it('returns the correct response', async () => {
-    Axios.get.mockReturnValue(Promise.resolve({ data: { status: 200 } }));
-    const response = await api.getCompaniesHouseDetails({ companyRegNumber: companiesHouseNumber, userToken });
-    expect(response).toEqual({ status: 200 });
+  const portalApiGetCompanyResponse = {
+    companiesHouseRegistrationNumber: MOCK_COMPANY_REGISTRATION_NUMBERS.VALID,
+    companyName: 'TEST COMPANY LTD',
+    registeredAddress: {
+      addressLine1: '1 Test Street',
+      locality: 'Test City',
+      postalCode: 'A1 2BC',
+      country: 'United Kingdom',
+    },
+    industries: [],
+  };
+
+  it('returns the company if it is returned by the request to Portal API', async () => {
+    Axios.get.mockResolvedValueOnce({ status: 200, data: portalApiGetCompanyResponse });
+
+    const response = await api.getCompanyByRegistrationNumber({ registrationNumber, userToken });
+
+    expect(response).toEqual({ company: portalApiGetCompanyResponse });
   });
 
-  it('throws an error if there is an api error', async () => {
-    Axios.get.mockReturnValue(Promise.reject());
-    await expect(api.getCompaniesHouseDetails({ companyRegNumber: companiesHouseNumber, userToken })).rejects.toThrowError();
+  it('returns the correct error information if it is called without a registration number', async () => {
+    const response = await api.getCompanyByRegistrationNumber({ userToken });
+
+    expect(response).toEqual({
+      errRef: 'regNumber',
+      errMsg: 'Enter a Companies House registration number',
+    });
   });
 
-  it('throws an appropriate error when given an invalid companiesHouseNumber', async () => {
-    await expect(api.getCompaniesHouseDetails({ companyRegNumber: 'invalid', userToken })).rejects.toThrowError('Invalid company house number');
+  it('returns the correct error information if it is called with a null registration number', async () => {
+    const response = await api.getCompanyByRegistrationNumber({ registrationNumber: null, userToken });
+
+    expect(response).toEqual({
+      errRef: 'regNumber',
+      errMsg: 'Enter a Companies House registration number',
+    });
+  });
+
+  it('returns the correct error information if it is called with a registration number that is the empty string', async () => {
+    const response = await api.getCompanyByRegistrationNumber({ registrationNumber: '', userToken });
+
+    expect(response).toEqual({
+      errRef: 'regNumber',
+      errMsg: 'Enter a Companies House registration number',
+    });
+  });
+
+  it('returns the correct error information if it is called with an invalid registration number', async () => {
+    const response = await api.getCompanyByRegistrationNumber({ registrationNumber: MOCK_COMPANY_REGISTRATION_NUMBERS.INVALID_TOO_SHORT, userToken });
+
+    expect(response).toEqual({
+      errRef: 'regNumber',
+      errMsg: 'Enter a valid Companies House registration number',
+    });
+  });
+
+  it.each([
+    {
+      status: 400,
```

Comment:
* As a variable please
* HttpStatusCode for HTTP status codes please.

Reply:
Done.